manticore/core/smtlib/solver.py

```diff
@@ -192,6 +222,11 @@ def start(self):
             close_fds=True,
         )
 
+        # stdout should be non-blocking
+        fl = fcntl.fcntl(self._proc.stdout, fcntl.F_GETFL)
```

Comment:
I won't pretend to have a comprehensive grasp of low-level IO in Python, but I'm more used to seeing the `select` module used than `fcntl`. Did you consider `select.poll`, which is also non-blocking, as a way to handle these file descriptors?

Reply:
I selected `fcntl` instead of `poll` because I did not wanted to add another thread/currency primitive for workers, which already works using threads..